crates/ruma-events/src/room/name.rs

```diff
@@ -47,15 +48,43 @@ where
     match Option::<String>::deserialize(deserializer)? {
         Some(name) => match name.len() {
             0 => Ok(None),
-            1..=255 => Ok(Some(name)),
+            1..=255 => match RoomName::try_from(name) {
+                Ok(name) => Ok(Some(name)),
+                Err(e) => Err(D::Error::custom(e.to_string())),
+            },
             _ => Err(D::Error::custom("a room name cannot be more than 255 bytes")),
         },
         None => Ok(None),
     }
 }
 
+/// The name of a room.
+///
+/// It should not exceed 255 characters.
+#[derive(Clone, Debug, Deserialize, Serialize, PartialEq, Eq)]
```

Comment:
I think it might be good to have `#[serde(transparent)]` here for the derived `Serialize` impl (it should then be a bit simpler, even though the outcome will be the same for JSON).

The `Deserialize` impl should not be derived, otherwise it's possible to create an empty or too long `RoomName` that way.